pkg/agent/baker.go

```
@@ -1005,6 +1005,22 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 		"InsertIMDSRestrictionRuleToMangleTable": func() bool {
 			return config.InsertIMDSRestrictionRuleToMangleTable
 		},
+		"ShouldEnableLocalDNS": func() bool {
+			return profile.ShouldEnableLocalDNS()
+		},
+		"GetGeneratedLocalDNSCoreFile": func() string {
+			output, err := GenerateLocalDNSCoreFile(config, profile, localDNSCoreFileTemplateString)
+			if err != nil {
+				panic(err)
```

I think we can just return the error like line 970 `GetSysctlContent` above. panic will cause the app to crash which is not desired in this case.

---

ok sure, made the change.